pkg/apis/garden/validation/validation.go

```diff
@@ -1530,6 +1530,30 @@ func validateKubernetes(kubernetes garden.Kubernetes, fldPath *field.Path) field
 		}
 	}
 
+	allErrs = append(allErrs, validateKubeControllerManager(kubernetes.KubeControllerManager, fldPath.Child("kubeControllerManager"))...)
+
+	return allErrs
+}
+
+func validateKubeControllerManager(kcm *garden.KubeControllerManagerConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if kcm != nil {
+		if hpa := kcm.HorizontalPodAutoscalerConfig; hpa != nil {
+			fldPath = fldPath.Child("horizontalPodAutoscaler")
+			if hpa.DownscaleDelay < 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("downscaleDelay"), hpa.DownscaleDelay, "downscale delay must not be negative"))
+			}
+			if hpa.SyncPeriod < 1*time.Second {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("syncPeriod"), hpa.SyncPeriod, "syncPeriod must not be less than a second"))
+			}
+			if hpa.Tolerance < 0.1 || hpa.Tolerance > 0.5 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("tolerance"), hpa.Tolerance, "tolerance of must be between 0.1 and 0.5"))
```

Comment:
Why do we not allow tolerances > 0.5?

Reply:
Because tolerance of 0.5 means that the possible desired values in the HPA spec have to be less than 50%.

But I am fine relaxing this limit. It is the user's prerogative. Perhaps we can just validate that it is non-negative?

Reply:
Not sure, I've not spend any time yet to dig into how the HPA works itself. I just was curious about why you've put this validation. I'm fine with keeping it if you say it makes sense. Maybe just add a comment there.

Reply:
On second thought, may be this limit of tolerance of 0.5 might not be suitable for custom metrics.

I will remove this limit and just validate for non-negative and non-zero values.

Reply:
Done.

Reply:
Thank you!